mysticeti-core/src/committee.rs

```diff
@@ -127,9 +129,28 @@ impl Committee {
         total_stake
     }
 
-    // TODO: fix to select by stake
-    pub fn elect_leader(&self, r: u64) -> AuthorityIndex {
-        (r % self.authorities.len() as u64) as AuthorityIndex
+    pub fn elect_leader(&self, round: u64) -> AuthorityIndex {
```

Comment:
Up to you, but I want to point out that you don't really need cfg-if dependency, the standard way to do this in rust is to just define two versions of function, one for test and one for non-test

Reply:
Thanks for pointing out Andrey! I actually chose to use this approach explicitly to avoid having two separate versions of the function and make it more obvious to the reader the whole logic in one place for the test and non-test case. In my experience it is easy to be missed otherwise and that's a delicate part of the logic. To be honest I would prefer not even doing that and as the TODO states have the leader election logic injected via a separate component which can swap/user according to each case.